airbyte-webapp/src/views/Connector/ServiceForm/components/Controls/ConnectorServiceTypeControl/useAnalyticsTrackFunctions.tsx

```diff
@@ -0,0 +1,33 @@
+import { capitalize } from "lodash";
+
+import { Action, Namespace } from "core/analytics";
+import { useAnalyticsService } from "hooks/services/Analytics";
+
+export const useAnalyticsTrackFunctions = (formType: "source" | "destination") => {
+  const analytics = useAnalyticsService();
+
+  const namespaceType = formType === "source" ? Namespace.SOURCE : Namespace.DESTINATION;
+
+  const trackMenuOpen = () => {
+    analytics.track(namespaceType, Action.SELECTION_OPENED, {
+      actionDescription: "Opened connector type selection",
+    });
+  };
+
+  const trackNoOptionMessage = (inputValue: string) => {
+    analytics.track(namespaceType, Action.NO_MATCHING_CONNECTOR, {
+      actionDescription: "Connector query without results",
+      query: inputValue,
+    });
+  };
+
+  const trackConnectorSelection = (connectorId: string, connectorName: string) => {
+    analytics.track(namespaceType, Action.SELECT, {
+      actionDescription: `${capitalize(formType)} connector type selected`,
+      [`connector_${formType}`]: connectorName,
+      [`connector_${formType}_definition_id`]: connectorId,
+    });
+  };
```

Comment:
If we go the route of hookifying those calls, I think we should `useCallback` them all, since those function might easily needed to be called inside another method in the consuming, which itself might want to use memoization. Having those methods here not memoized would cause any function consuming them to also not to be memooizable, but it might not be obvious by just looking at the component.

Reply:
Duly noted! Will update the tracking functions